bot/exts/utilities/color.py

```diff
@@ -0,0 +1,224 @@
+        for colour_space, value in colour_conversions.items():
+            colour_embed.add_field(
+                name=colour_space,
+                value=f"`{value}`",
+                inline=True
+            )
+
+        thumbnail = Image.new("RGB", THUMBNAIL_SIZE, color=rgb)
+        buffer = BytesIO()
+        thumbnail.save(buffer, "PNG")
+        buffer.seek(0)
+        thumbnail_file = File(buffer, filename="colour.png")
+
+        colour_embed.set_thumbnail(url="attachment://colour.png")
+
+        await ctx.send(file=thumbnail_file, embed=colour_embed)
+
+    @commands.group(aliases=("color",))
+    async def colour(self, ctx: commands.Context) -> None:
+        """User initiated command to create an embed that displays colour information."""
+        if ctx.invoked_subcommand is None:
+            await invoke_help_command(ctx)
+
+    @colour.command()
+    async def rgb(self, ctx: commands.Context, red: int, green: int, blue: int) -> None:
+        """Command to create an embed from an RGB input."""
+        if any(c not in range(0, 256) for c in (red, green, blue)):
+            raise BadArgument(
+                message=f"RGB values can only be from 0 to 255. User input was: `{red, green, blue}`."
+            )
+        rgb_tuple = (red, green, blue)
+        await self.send_colour_response(ctx, rgb_tuple)
+
+    @colour.command()
+    async def hsv(self, ctx: commands.Context, hue: int, saturation: int, value: int) -> None:
+        """Command to create an embed from an HSV input."""
+        if (hue not in range(0, 361)) or any(c not in range(0, 101) for c in (saturation, value)):
+            raise BadArgument(
+                message="Hue can only be from 0 to 360. Saturation and Value can only be from 0 to 100. "
+                f"User input was: `{hue, saturation, value}`."
+            )
+        # ImageColor HSV expects S and V to be swapped
+        hsv_tuple = ImageColor.getrgb(f"hsv({hue}, {value}%, {saturation}%)")
+        await self.send_colour_response(ctx, hsv_tuple)
+
+    @colour.command()
+    async def hsl(self, ctx: commands.Context, hue: int, saturation: int, lightness: int) -> None:
+        """Command to create an embed from an HSL input."""
+        if (hue not in range(0, 361)) or any(c not in range(0, 101) for c in (saturation, lightness)):
+            raise BadArgument(
+                message="Hue can only be from 0 to 360. Saturation and Lightness can only be from 0 to 100. "
+                f"User input was: `{hue, saturation, lightness}`."
+            )
+        hsl_tuple = ImageColor.getrgb(f"hsl({hue}, {saturation}%, {lightness}%)")
+        await self.send_colour_response(ctx, hsl_tuple)
+
+    @colour.command()
+    async def cmyk(self, ctx: commands.Context, cyan: int, magenta: int, yellow: int, key: int) -> None:
+        """Command to create an embed from a CMYK input."""
+        if any(c not in range(0, 101) for c in (cyan or magenta or yellow or key)):
```

Comment:
You should be iterating through a tuple here
```suggestion
        if any(c not in range(0, 101) for c in (cyan, magenta, yellow, key)):
```